paperqa/agents/search.py

```diff
@@ -399,10 +428,12 @@ async def query(
         addresses = [
             s[1]
             for s in searcher.search(
-                index.parse_query(self.clean_query(query), query_fields), top_n
+                index.parse_query(self.clean_query(query), query_fields),
+                top_n,
+                offset=offset,
             ).hits
             if s[0] > min_score
-        ][offset : offset + top_n]
+        ]
```

Comment:
This is a bugfix. We were getting top_n and then starting at offset.

Reply:
I think the change is good, but it's orthogonal to RAG QA arena.

Can you (1) make a unit test for this and (2) do it in a separate PR?

Reply:
https://github.com/Future-House/paper-qa/pull/846